malwarecage/resources/quick_query.py

```diff
@@ -0,0 +1,110 @@
+                description: List of saved queries attached to the type of objects view.
+                content:
+                  application/json:
+                    schema:
+                      type: array
+                      items: QuickQueryResponseSchema
+            400:
+                description: When query is invalid
+        """
+        schema = QuickQuerySchemaBase()
+        obj = schema.loads(request.get_data(as_text=True))
+        if obj.errors:
+            return {"errors": obj.errors}, 400
+        quick_query = QuickQuery(
+            query=obj.data["query"],
+            name=obj.data["name"],
+            type=type,
+            user_id=g.auth_user.id,
+        )
+        db.session.add(quick_query)
+        db.session.commit()
+        logger.info('Query saved', extra={'query': quick_query.name})
+
+        db.session.refresh(quick_query)
+        schema = QuickQueryResponseSchema()
+        return schema.dump(quick_query)
+
+    @requires_authorization
+    def get(self, type):
+        """
+        ---
+        summary: Get stored quick queries
+        description: Returns all saved queries related with specified object type.
+        tags:
+            - query
+        parameters:
+            - in: path
+              name: type
+              schema:
+                type: string
+                enum: [file, config, blob, object]
+              description: Type of objects view.
+        responses:
+            200:
+                description: List of saved queries attached to the type of objects view.
+                content:
+                  application/json:
+                    schema:
+                      type: array
+                      items: QuickQueryResponseSchema
+        """
+        quick_queries = db.session.query(QuickQuery).filter(QuickQuery.type == type).filter(QuickQuery.user_id == g.auth_user.id).all()
+        schema = QuickQueryResponseSchema(many=True)
+        return schema.dump(quick_queries)
+
+
+class QuickQueryItemResource(Resource):
+    @requires_authorization
+    def delete(self, id):
+        """
+        ---
+        summary: Delete query
+        description: Delete custom quick query.
+        tags:
+            - query
+        parameters:
+            - in: path
+              name: id
+              schema:
+                id: int
+              description: Query identifier
+        responses:
+            404:
+                description: When query was not found.
+            200:
+                description: When query was successfully deleted
+        """
+        quick_query = db.session.query(QuickQuery).filter(QuickQuery.id == id).filter(QuickQuery.user_id == g.auth_user.id).first()
+        if quick_query is None:
+            raise NotFound("Query was not found")
+        if quick_query is not None:
+            db.session.delete(quick_query)
+            logger.info('query deleted', extra={'query': id})
+            db.session.commit()
```

Comment:
`quick_query is not None` condition is always true here
```suggestion

        db.session.delete(quick_query)
        logger.info('Query deleted', extra={'query': id})
        db.session.commit()
```